### ISSUE TITLE
Solutions in examples

### ISSUE THREAD

**Issue by oscarlevin** [rbeezer/mathbook#248]
It would be nice to be able to give solutions inside examples.  This is different than an exercise, which implies that the student should attempt to solve the question (and would be something that could be gathered into a workbook).  Not all examples will be examples of how to solve problems, but many will and modeling the question/solution format would be nice.

**Comment by oscarlevin** [comment]
Here is the proposed structure.  An example could just have content, as it is now.  Alternatively, there could be a <statement> and a <solution>, in which case the content of the solution would be knowled for html.  For the latex, this would just be presented as a new line, preferably with a bold "Solution" starting it off.  There could also be a hint, although I would be happy to just have the hint as part of the statement.  An answer would probably not be needed; the point of this sort of example is to illustrate how to solve a problem.

**Comment by rplantz** [comment]
I thought about this as I'm writing my book. I realized that, from my student days (half a century ago), I'm used to examples being done for me by the author, and exercises being problems, some of which were assigned by the professor. Usually, I only did the assigned exercises.

In my current writing, I decided that I want the reader to treat exercises as, well, a means to exercise. So I use them as a way to extend my explanations and worked-out examples. I'm hoping this will encourage readers to work all the exercises and see them as an essential part of reading the text. I think this might work better with the HTML/knowl presentation. I'm no longer teaching, so I'll have to rely on feedback to know how well this actually works in practice.

I should also mention that my book is a computer science text on programming, and it's very hands-on.
